multi-transfer-esdt/src/lib.rs

```diff
@@ -33,55 +42,64 @@ pub trait MultiTransferEsdt:
     fn batch_transfer_esdt_token(
         &self,
         #[var_args] transfers: ManagedVarArgs<SingleTransferTuple<Self::Api>>,
-    ) -> ManagedMultiResultVec<TransactionStatus> {
-        let mut tx_statuses = ManagedVec::new();
+    ) {
         let mut cached_token_ids = ManagedVec::new();
         let mut cached_prices = ManagedVec::new();
 
         for transfer in transfers {
-            let to = &transfer.address;
-            let token_id = &transfer.token_id;
-            let amount = &transfer.amount;
-
-            if to.is_zero() || self.blockchain().is_smart_contract(to) {
-                tx_statuses.push(TransactionStatus::Rejected);
+            if transfer.to.is_zero() || self.blockchain().is_smart_contract(&transfer.to) {
+                self.add_refund_tx_to_batch(transfer);
                 continue;
             }
-            if !self.token_whitelist().contains(token_id)
-                || !self.is_local_role_set(token_id, &EsdtLocalRole::Mint)
+            if !self.token_whitelist().contains(&transfer.token_id)
+                || !self.is_local_role_set(&transfer.token_id, &EsdtLocalRole::Mint)
             {
-                tx_statuses.push(TransactionStatus::Rejected);
+                self.add_refund_tx_to_batch(transfer);
                 continue;
             }
 
             let queried_fee: BigUint;
-            let required_fee = match cached_token_ids.iter().position(|id| &id == token_id) {
+            let required_fee = match cached_token_ids
```

Comment:
with the new version we can make transaction fees free on elrond - so this can be deleted - or should work with 0 :)

Reply:
This works with 0 fees if price_aggregator_address is set to the zero-address, but I'll remove this section in a next PR.